tests/utils/network_setup.go

```diff
@@ -0,0 +1,275 @@
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"os"
+
+	"github.com/ava-labs/avalanche-network-runner/rpcpb"
+	"github.com/ava-labs/avalanchego/ids"
+	relayerEvm "github.com/ava-labs/awm-relayer/vms/evm"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ava-labs/subnet-evm/plugin/evm"
+	"github.com/ava-labs/subnet-evm/rpc"
+	"github.com/ava-labs/subnet-evm/tests/utils/runner"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	fundedKeyStr = "56289e99c94b6912bfc12adc093c9b51124f0dc54ac7a766b2bc5ccf558d8027"
+)
+
+var (
+	teleporterContractAddress      common.Address
+	subnetA, subnetB               ids.ID
+	blockchainIDA, blockchainIDB   ids.ID
+	chainANodeURIs, chainBNodeURIs []string
+	fundedAddress                  = common.HexToAddress("0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC")
+	fundedKey                      *ecdsa.PrivateKey
+	chainAWSClient, chainBWSClient ethclient.Client
+	chainAWSURI, chainBWSURI       string
+	chainAIDInt, chainBIDInt       *big.Int
+
+	// Internal vars only used to set up the local network
+	anrConfig           = runner.NewDefaultANRConfig()
+	manager             = runner.NewNetworkManager(anrConfig)
+	warpChainConfigPath string
+)
+
+//
+// Global test state getters. Should be called within a test spec, after SetupNetwork has been called
+//
+
+type SubnetTestInfo struct {
+	SubnetID      ids.ID
+	BlockchainID  ids.ID
+	ChainNodeURIs []string
+	ChainWSClient ethclient.Client
+	ChainWSURI    string
+	ChainIDInt    *big.Int
+}
+
+func GetSubnetsInfo() []SubnetTestInfo {
+	return []SubnetTestInfo{
+		GetSubnetATestInfo(),
+		GetSubnetBTestInfo(),
+	}
+}
+
+func GetSubnetATestInfo() SubnetTestInfo {
+	return SubnetTestInfo{
+		SubnetID:      subnetA,
+		BlockchainID:  blockchainIDA,
+		ChainNodeURIs: chainANodeURIs,
+		ChainWSClient: chainAWSClient,
+		ChainWSURI:    chainAWSURI,
+		ChainIDInt:    big.NewInt(0).Set(chainAIDInt),
+	}
+}
+func GetSubnetBTestInfo() SubnetTestInfo {
+	return SubnetTestInfo{
+		SubnetID:      subnetB,
+		BlockchainID:  blockchainIDB,
+		ChainNodeURIs: chainBNodeURIs,
+		ChainWSClient: chainBWSClient,
+		ChainWSURI:    chainBWSURI,
+		ChainIDInt:    big.NewInt(0).Set(chainBIDInt),
+	}
+}
+func GetTeleporterContractAddress() common.Address {
+	return teleporterContractAddress
+}
+func GetFundedAccountInfo() (common.Address, *ecdsa.PrivateKey) {
+	key, err := crypto.ToECDSA(crypto.FromECDSA(fundedKey))
+	Expect(err).Should(BeNil())
+	return fundedAddress, key
+}
+
+// setupNetwork starts the default network and adds 10 new nodes as validators with BLS keys
+// registered on the P-Chain.
+// Adds two disjoint sets of 5 of the new validator nodes to validate two new subnets with a
+// a single Subnet-EVM blockchain.
+func SetupNetwork(warpGenesisFile string) {
```

Comment:
Super nit: I think this should be `SetUpNetwork`. That would also make it consistent with `SetUpProposerVM`

Reply:
If it's all the same to you, I'm going to instead change it to `SetupProposerVM`. That way we're consistent with the `subnet-evm` `SetupNetwork` function call made by this function.

Reply:
😅 I didn't notice that `SetupNetwork` was a `subnet-evm` function. The grammar police in me wants to say that's not right in this case since we're using it as a verb (set up) and not a noun (setup), but lets keep it for the sake of consistency haha.

Reply:
Could you just capitalize "SetupNetwork" in the comment above the function now that it's exported?